play/src/components/joingame/EnterGameCode.tsx

```diff
@@ -0,0 +1,98 @@
+import React, {useState} from 'react';
+import { styled, useTheme } from '@mui/material/styles';
+import { Stack, Box, Typography } from '@mui/material';
+import {
+  IntroButton,
+  IntroTextField,
+  JoinGameBackgroundContainer,
+} from '../../lib/styledcomponents/StyledComponents';
+import Logo from '../../img/rightOnLogo.svg';
+
+const StackContainer = styled(Stack)(({ theme }) => ({
+  display: 'flex',
+  justifyContent: 'center',
+  alignItems: 'center',
+  maxWidth: theme.breakpoints.values.xs,
+}));
+
+const PaddedContainer = styled(Box)(({ theme }) => ({
+  paddingLeft: `${theme.sizing.smallPadding}px`,
+  paddingRight: `${theme.sizing.smallPadding}px`,
+}));
+
+interface EnterGameCodeProps {
+  gameCodeValue: string;
+  setGameCodeValue: (newValue: string) => void;
+  inputError: boolean;
+}
+
+export default function EnterGameCode({
+  gameCodeValue,
+  setGameCodeValue,
+  inputError,
+}: EnterGameCodeProps) {
+  const theme = useTheme();
+  const [value, setValue] = useState('');
+  return (
+    <JoinGameBackgroundContainer>
+      <StackContainer spacing={5}>
+        <img
+          style={{
+            width: '214px',
+            height: '118px',
+            paddingTop: `${theme.sizing.extraLargePadding}px`,
+          }}
+          src={Logo}
+          alt="Question"
+        />
+        {/* container here to trim the spacing set by parent stack between text and input, typ */}
+        <Box>
+          <Typography variant="h2" sx={{ weight: 700, textAlign: 'center' }}>
+            Enter Game Code
+          </Typography>
+          <IntroTextField
+            fullWidth
+            variant="filled"
+            autoComplete="off"
+            placeholder="####"
+            onChange={(event) =>  setGameCodeValue(event.target.value)}
```

Comment:
nit: double space before `setGameCodeValue`

Reply:
I haven't looked into `IntroTextField` so this might already be managed there, but if it's not, we should to make sure this field only accepts numbers, I believe there're also attributes to set to help mobile web to bring numeric keyboard instead of the generic one

Reply:
This has been updated. The numeric keyboard is brought up by adding `inputMode: 'numeric'` to `inputProps.` Restricting the input is more complicated though. The docs (https://mui.com/material-ui/react-text-field/) not that `type='number'` is flawed and provide a flag that only returns an invalid state (doesn't restrict input). Currently, I'm just parsing the input myself on the onChange event and it seems to work well. A comment with a link to the docs has been added (so we can be reminded to migrate to the numeric input if they ever release it)